File: GNSDK/wrappers/gnsdk_csharp/src_wrapper/list_element_child_provider.cs

namespace GracenoteSDK {

using System;
using System.Runtime.InteropServices;

public class list_element_child_provider : IDisposable {
  private HandleRef swigCPtr;
  protected bool swigCMemOwn;

  internal list_element_child_provider(IntPtr cPtr, bool cMemoryOwn) {
    swigCMemOwn = cMemoryOwn;
    swigCPtr = new HandleRef(this, cPtr);
  }

  internal static HandleRef getCPtr(list_element_child_provider obj) {
    return (obj == null) ? new HandleRef(null, IntPtr.Zero) : obj.swigCPtr;
  }

  ~list_element_child_provider() {
    Dispose();
  }

  public virtual void Dispose() {
    lock(this) {
      if (swigCPtr.Handle != IntPtr.Zero) {
        if (swigCMemOwn) {
          swigCMemOwn = false;
          gnsdk_csharp_marshalPINVOKE.delete_list_element_child_provider(swigCPtr);
        }
        swigCPtr = new HandleRef(null, IntPtr.Zero);
      }
      GC.SuppressFinalize(this);
    }
  }

/** 
* Constructor
*/
  public list_element_child_provider() : this(gnsdk_csharp_marshalPINVOKE.new_list_element_child_provider(), true) {
  }

/** 
* Get list elment by position
*/
  public GnListElement get_data(uint pos) {
    GnListElement ret = new GnListElement(gnsdk_csharp_marshalPINVOKE.list_element_child_provider_get_data(swigCPtr, pos), true);
    return ret;
  }

/** 
* Get child list element count
*/
  public uint count() {
    uint ret = gnsdk_csharp_marshalPINVOKE.list_element_child_provider_count(swigCPtr);
    return ret;
  }

}

}

File: GNSDK/wrappers/gnsdk_java/src_wrapper/j2se/com/gracenote/gnsdk/list_element_child_provider.java

package com.gracenote.gnsdk;

public class list_element_child_provider {
  private transient long swigCPtr;
  protected transient boolean swigCMemOwn;

  protected list_element_child_provider(long cPtr, boolean cMemoryOwn) {
    swigCMemOwn = cMemoryOwn;
    swigCPtr = cPtr;
  }

  protected static long getCPtr(list_element_child_provider obj) {
    return (obj == null) ? 0 : obj.swigCPtr;
  }

  protected void finalize() {
    delete();
  }

  public synchronized void delete() {
    if (swigCPtr != 0) {
      if (swigCMemOwn) {
        swigCMemOwn = false;
        gnsdk_javaJNI.delete_list_element_child_provider(swigCPtr);
      }
      swigCPtr = 0;
    }
  }

/**  
* Constructor 
*/ 
 
  public list_element_child_provider() {
    this(gnsdk_javaJNI.new_list_element_child_provider(), true);
  }

  public GnListElement getData(long pos) {
    return new GnListElement(gnsdk_javaJNI.list_element_child_provider_getData(swigCPtr, this, pos), true);
  }

/**  
* Get child list element count 
*/ 
 
  public long count() {
    return gnsdk_javaJNI.list_element_child_provider_count(swigCPtr, this);
  }

}
